identity_credential/src/credential/domain_linkage_configuration.rs

```diff
@@ -0,0 +1,160 @@
+///   is supported.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct DomainLinkageConfiguration {
+  /// Fixed context.
+  #[serde(rename = "@context")]
+  context: Context,
+  /// Linked credentials.
+  linked_dids: Vec<Credential>,
+  /// Further properties, must be empty.
+  #[serde(flatten)]
+  pub properties: Object,
+}
+
+impl Display for DomainLinkageConfiguration {
+  fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+    self.fmt_json(f)
+  }
+}
+
+impl DomainLinkageConfiguration {
+  /// Creates a new DID Configuration Resource.
+  pub fn new(linked_dids: Vec<Credential>) -> Self {
+    Self {
+      context: Self::well_known_context().clone(),
+      linked_dids,
+      properties: Object::new(),
+    }
+  }
+
+  pub(crate) fn well_known_context() -> &'static Context {
+    &WELL_KNOWN_CONTEXT
+  }
+
+  pub(crate) const fn domain_linkage_type() -> &'static str {
+    "DomainLinkageCredential"
+  }
+
+  /// Validates the semantic structure.
+  pub fn check_structure(&self) -> Result<()> {
+    if &self.context != Self::well_known_context() {
+      return Err(DomainLinkageError("invalid context".into()));
+    }
+
+    if !self.properties.is_empty() {
+      return Err(DomainLinkageError(
+        "domain linkage configuration contains invalid properties".into(),
+      ));
+    }
+
+    if self.linked_dids.is_empty() {
+      return Err(DomainLinkageError("property `linked_dids` is invalid".into()));
+    }
+    Ok(())
+  }
+
+  /// Fetches the the DID Configuration resource via a GET request at the
+  /// well-known location: "`origin`/.well-known/did-configuration.json".
+  #[cfg(feature = "domain-linkage-fetch")]
+  pub async fn fetch_configuration(mut origin: Url) -> Result<DomainLinkageConfiguration> {
+    if origin.scheme() != "https" {
+      return Err(DomainLinkageError("origin` does not use `https` protocol".into()));
+    }
+    // todo: set file size limit to 1MB.
+    let client: Client = reqwest::ClientBuilder::new()
+      .https_only(true)
+      .redirect(Policy::none())
+      .build()
+      .map_err(|err| DomainLinkageError(Box::new(err)))?;
+
+    origin.set_path(".well-known/did-configuration.json");
+    let domain_linkage_configuration: DomainLinkageConfiguration = client
+      .get(origin.to_string())
+      .send()
+      .await
+      .map_err(|err| DomainLinkageError(Box::new(err)))?
+      .json::<DomainLinkageConfiguration>()
+      .await
+      .map_err(|err| DomainLinkageError(Box::new(err)))?;
+
+    //todo: should structure be checked here? it will be inconsistent with `from_json(..)`.
+    Ok(domain_linkage_configuration)
+  }
+
+  /// List of Domain Linkage Credentials.
+  pub fn linked_dids(&self) -> &Vec<Credential> {
+    &self.linked_dids
+  }
+
+  /// List of the issuers of the Domain Linkage Credentials.
+  pub fn issuers(&self) -> Vec<Url> {
+    self
+      .linked_dids
+      .iter()
+      .map(|linked_did| linked_did.issuer.url().clone())
+      .collect()
+  }
```

Comment:
I think Rust users would typically prefer an iterator to be returned here: 
```suggestion
  pub fn issuers(&self) -> impl Iterator<Item = &Url> {
    self
      .linked_dids
      .iter()
      .map(|linked_did| linked_did.issuer.url())
  }
```